### PR TITLE
Database: Items, Visited Maps/Taxis, Trophies, Mails

### PR DESCRIPTION
**Pull request by AngeloTadeucci**
- Fixed `RequestTutorialItemHandler`;
- Moved char decorations (hair, eye, etc) to cosmetics;
- BCrypt hash passwords;
- Fixed VIP after relogging;
- Fixed taxi price;
- Fixed `ItemInventoryPacket`;
- Fixed `StorageInventory`, renamed to `StorageInventoryPacket`;
- Adjusted `ServerEnterPacket`;

### CODE REVIEW

File: MapleServer2/PacketHandlers/Game/MoveFieldHandler.cs [hunk 1]
```diff
@@ -44,6 +44,7 @@ public override void Handle(GameSession session, PacketReader packet)
                     session.Player.Rotation = session.FieldPlayer.Rotation;
                     session.Player.Coord = session.Player.ReturnCoord;
                     session.Player.ReturnCoord.Z += Block.BLOCK_SIZE;
+                    DatabaseManager.UpdateCharacter(session.Player);
```
Comment (AlanMorel) [inline MapleServer2/PacketHandlers/Game/MoveFieldHandler.cs:47]: just curious, any reason we need to save to database upon a map change?

File: MapleServer2/PacketHandlers/Login/LoginHandler.cs [hunk 2]
```diff
@@ -38,17 +39,20 @@ public override void Handle(LoginSession session, PacketReader packet)
             string username = packet.ReadUnicodeString();
             string password = packet.ReadUnicodeString();
 
+            // Creates md5 hash of password
```
Comment (AlanMorel) [inline MapleServer2/PacketHandlers/Login/LoginHandler.cs:42]: No bcrypt in 2021?

File: MapleServer2/Types/Item.cs [hunk 3]
```diff
@@ -186,5 +175,27 @@ public static bool IsPet(int itemId)
         {
             return itemId >= 60000001 && itemId < 61000000;
         }
+
+        public void GetMetadataValues(int id)
```
Comment (AlanMorel) [inline MapleServer2/Types/Item.cs:179]: this isn't really getting as much as it is setting, perhaps rename?

**Review by AlanMorel — CHANGES_REQUESTED** [review]

File: MapleServer2/PacketHandlers/Game/MoveFieldHandler.cs (hunk 1 above)
Comment (AngeloTadeucci) [reply MapleServer2/PacketHandlers/Game/MoveFieldHandler.cs:47]: We don't need to save on map change but I'm using as one way to manually test if changes are being saved. I could remove these, and add a command to save and implement a task to save every 10~20 minutes?

**Review by AlanMorel — CHANGES_REQUESTED** [review]

**Review by Sparkymod — APPROVED** [review]

File: MapleServer2/PacketHandlers/Game/MoveFieldHandler.cs (hunk 1 above)
Comment (AlanMorel) [reply MapleServer2/PacketHandlers/Game/MoveFieldHandler.cs:47]: As long as this is the only place we are saving unnecessarily, then yeah it's fine

**Pull request merged**